storage/remote/client.go

```diff
@@ -195,10 +230,15 @@ func (c *Client) Read(ctx context.Context, query *prompb.Query) (*prompb.QueryRe
 
 	compressed, err = ioutil.ReadAll(httpResp.Body)
 	if err != nil {
+		remoteReadRequestTotalCounter = remoteReadQueriesRequestTotal.WithLabelValues(c.remoteName, c.url.String(), "504")
```

Comment:
Above you've already called `remoteReadRequestTotalCounter := remoteReadQueriesRequestTotal.WithLabelValues(c.remoteName, c.url.String(), strconv.Itoa(httpResp.StatusCode))`, which is correct. We've got an HTTP response which should have an error code. The only thing to change up there is that we need to check `if err != nil` before we access `httpResp` to get the status code. The incrementing of the counter can also be done above.

Here (and below) we shouldn't have to make add any instrumentation code, errors with reading the body, decoding, etc. are separate from the metric that just counts the # of remote read queries executed.